plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2147,12 +2147,14 @@ public Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(
             Collection<ComputedStatistics> computedStatistics,
             List<ConnectorTableHandle> sourceTableHandles)
     {
-        // delete before insert .. simulating overwrite
-        executeDelete(session, tableHandle);
-
         IcebergWritableTableHandle table = (IcebergWritableTableHandle) insertHandle;
 
         Table icebergTable = transaction.table();
+        // delete before insert .. simulating overwrite
+        transaction.newDelete()
+                .deleteFromRowFilter(Expressions.alwaysTrue())
```

Comment:
@raunaqmorarka I wasn't totally sure about this `alwaysTrue` part. Do we always rewrite the entire content of a MV?

Reply:
I think incremental refreshes are supported only if `io.trino.spi.connector.ConnectorMetadata#delegateMaterializedViewRefreshToConnector` returns true (it returns false for Iceberg), and connector knows how to refresh incrementally.

BTW @alexjo2144 will this unregister all old data files, or will it scan any data file contents?

Also, does this preserve table version history correctly?

Reply:
> BTW @alexjo2144 will this unregister all old data files, or will it scan any data file contents?

This will unregister the data files from the manifest and they will not be scanned. It is the same code as is run by `executeDelete` just attached to the transaction instead of being run separately. 

> Also, does this preserve table version history correctly?

If we need history to be retained we should upgrade to `0.14.1` first, there's a bug fix in that we'd need. However, I was assuming history for an MV is not useful

Reply:
> If we need history to be retained we should upgrade to `0.14.1` first, there's a bug fix in that we'd need

yeah, just realized it's coming now: https://github.com/trinodb/trino/pull/14074#issuecomment-1248522221

> However, I was assuming history for an MV is not useful

perhaps; i am fine merging as is, without 0.141